src/lib_ccx/ccx_encoders_spupng.c

```diff
@@ -727,10 +727,12 @@ int spupng_export_string2png(struct spupng_t *sp, char *str, FILE* output)
 	memset(buffer, 0, canvas_width * canvas_height * sizeof(struct pixel_t));
 
 	//str = "<font color=\"#66CDAA\"> __ This should be aquamarine </font> Regular <i> Italics font</i> <font color=\"#00ff00\">This should be green.</font> <u> Underlining __ testing </u> Regular text. Even more text. Random text. More text."; // Test string
-	char *tmp = (char*)malloc((strlen(str) + 1) * sizeof(char));
+	int len = strlen(str);
+	char *tmp = (char*)malloc((len + 1) * sizeof(char));
 	if (!tmp)
 		return -1;
-	strncpy(tmp, str, strlen(str)+1);
+	memcpy(tmp, str, len);
+	tmp[len] = '\0';
```

Comment:
What warnings is this fixing? strncpy() seems OK to me...
but if you are using memcpy then memcpy (tmp, str, len+1) would be better - you know that the source string does have the final 0 (otherwise strlen() would crash) so there's no need to add it manually.

Reply:
The warning is as follows: warning: ‘strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]

Reply:
The compiler complains because the 3rd parameter depends on the size of the source, instead of the destination.

Reply:
Consider just using strdup() instead.

Reply:
OK.